RDP brute force stops due to password expiry

**Describe the bug**
Attempting an RDP brute force will error out with `all children were disabled due too many connection errors`

The error seems to be due to `freerdp: The password has certainly expired and must be changed. (0x0002000f)`


**To Reproduce**
`hydra -L user.list -P creds.txt rdp://192.168.224.248 -t 1 -d`

**Expected behavior**
I would not expect a password expiration to cause hydra to stop brute forcing

**Desktop (please complete the following information):**
 - OS: Kali 2023.1
 - hydra version 9.4
 
**Additional context**
```
hydra -L user.list -P creds.txt rdp://192.168.224.248 -t 1 -d
Hydra v9.4 (c) 2022 by van Hauser/THC & David Maciejak - Please do not use in military or secret service organizations, or for illegal purposes (this is non-binding, these *** ignore laws and ethics anyway).

[DEBUG] Output color flag is 1
Hydra (https://github.com/vanhauser-thc/thc-hydra) starting at 2023-05-21 23:10:19
[DEBUG] cmdline: hydra -L user.list -P creds.txt -t 1 -d rdp://192.168.224.248
[DEBUG] opt:8 argc:9 mod:rdp tgt:192.168.224.248 port:0 misc:(null)
[WARNING] the rdp module is experimental. Please test, report - and if possible, fix.
[DATA] max 1 task per 1 server, overall 1 task, 25 login tries (l:5/p:5), ~25 tries per task
[DATA] attacking rdp://192.168.224.248:3389/
[VERBOSE] Resolving addresses ...
[DEBUG] resolving 192.168.224.248
[VERBOSE] resolving done
[DEBUG] Code: attack   Time: 1684725019
[DEBUG] Options: mode 3  ssl 0  restore 0  showAttempt 0  tasks 1  max_use 1 tnp 0  tpsal 0  tprl 0  exit_found 0  miscptr (null)  service rdp
[DEBUG] Brains: active 0  targets 1  finished 0  todo_all 25  todo 25  sent 0  found 0  countlogin 5  sizelogin 29  countpass 5  sizepass 74
[DEBUG] Target 0 - target 192.168.224.248  ip 192.168.224.248  login_no 0  pass_no 0  sent 0  pass_state 0  redo_state 0 (0 redos)  use_count 0  failed 0  done 0  fail_count 0  login_ptr zachary  pass_ptr welcome1
[DEBUG] Task 0 - pid 0  active 0  redo 0  current_login_ptr (null)  current_pass_ptr (null)
[DEBUG] Tasks 1 inactive  0 active
[DEBUG] child 0 got target 0 selected
[DEBUG] child 0 spawned for target 0 with pid 1050269
[DEBUG] head_no 0 has pid 1050269
[DEBUG] head_no[0] read n
[DEBUG] send_next_pair_init target 0, head 0, redo 0, redo_state 0, pass_state 0. loop_mode 0, curlogin (null), curpass (null), tlogin zachary, tpass welcome1, logincnt 0/5, passcnt 0/5, loop_cnt 1
[DEBUG] send_next_pair_mid done 1, pass_state 3, clogin zachary, cpass welcome1, tlogin zachary, tpass SomersetVinyl1!, redo 0
[ATTEMPT] target 192.168.224.248 - login "zachary" - pass "welcome1" - 1 of 25 [child 0] (0/0)
[DEBUG] rdp reported 0002000f
[ERROR] freerdp: The password has certainly expired and must be changed. (0x0002000f)
[DEBUG] pid 1050269 called child_exit with code 2
[DEBUG] head_no[0] read E
[ATTEMPT-ERROR] target 192.168.224.248 - login "zachary" - pass "welcome1" - child 0 - 1 of 25
[DEBUG] hydra_increase_fail_count: 1 >= 0 => disable
[DEBUG] - will be retried at the end: ip 192.168.224.248 - login zachary - pass welcome1 - child 0
[DEBUG] head_no 0, kill 1, fail 0
[DEBUG] child 0 got target 0 selected
[DEBUG] child 0 spawned for target 0 with pid 1050302
[DEBUG] head_no 0 has pid 1050302
[DEBUG] head_no[0] read n
[DEBUG] send_next_pair_init target 0, head 0, redo 1, redo_state 0, pass_state 3. loop_mode 0, curlogin , curpass , tlogin zachary, tpass SomersetVinyl1!, logincnt 0/5, passcnt 1/5, loop_cnt 1
[COMPLETED] target 192.168.224.248 - login "" - pass "" - child 0 - 1 of 26
[DEBUG] send_next_pair_mid done 1, pass_state 3, clogin zachary, cpass SomersetVinyl1!, tlogin zachary, tpass HabitsAgesEnd123, redo 1
[ATTEMPT] target 192.168.224.248 - login "zachary" - pass "SomersetVinyl1!" - 2 of 26 [child 0] (0/1)
[DEBUG] rdp reported 0002000f
[ERROR] freerdp: The password has certainly expired and must be changed. (0x0002000f)
[DEBUG] pid 1050302 called child_exit with code 2
[DEBUG] head_no[0] read E
[ATTEMPT-ERROR] target 192.168.224.248 - login "zachary" - pass "SomersetVinyl1!" - child 0 - 2 of 25
[DEBUG] hydra_increase_fail_count: 2 >= 0 => disable
[DEBUG] - will be retried at the end: ip 192.168.224.248 - login zachary - pass SomersetVinyl1! - child 0
[DEBUG] head_no 0, kill 1, fail 0
[DEBUG] child 0 got target 0 selected
[DEBUG] child 0 spawned for target 0 with pid 1050319
[DEBUG] head_no 0 has pid 1050319
[DEBUG] head_no[0] read n
[DEBUG] send_next_pair_init target 0, head 0, redo 2, redo_state 0, pass_state 3. loop_mode 0, curlogin , curpass , tlogin zachary, tpass HabitsAgesEnd123, logincnt 0/5, passcnt 2/5, loop_cnt 1
[COMPLETED] target 192.168.224.248 - login "" - pass "" - child 0 - 2 of 27
[DEBUG] send_next_pair_mid done 1, pass_state 3, clogin zachary, cpass HabitsAgesEnd123, tlogin zachary, tpass Luigi=Papal1963, redo 2
[ATTEMPT] target 192.168.224.248 - login "zachary" - pass "HabitsAgesEnd123" - 3 of 27 [child 0] (0/2)
[DEBUG] rdp reported 0002000f
[ERROR] freerdp: The password has certainly expired and must be changed. (0x0002000f)
[DEBUG] pid 1050319 called child_exit with code 2
[DEBUG] head_no[0] read E
[ATTEMPT-ERROR] target 192.168.224.248 - login "zachary" - pass "HabitsAgesEnd123" - child 0 - 3 of 25
[DEBUG] hydra_increase_fail_count: 3 >= 0 => disable
[DEBUG] - will be retried at the end: ip 192.168.224.248 - login zachary - pass HabitsAgesEnd123 - child 0
[DEBUG] head_no 0, kill 1, fail 2
[ERROR] all children were disabled due too many connection errors
[DEBUG] while loop left with -1
0 of 1 target completed, 0 valid password found
[DEBUG] killing all remaining children now that might be stuck
Hydra (https://github.com/vanhauser-thc/thc-hydra) finished at 2023-05-21 23:10:24
```

Im running into the exact same issue with hydra v9.4 on Kali.
`└─$ sudo hydra -vV -L users.txt -P passwords.txt -S rdp://192.168.196.248
Hydra v9.4 (c) 2022 by van Hauser/THC & David Maciejak - Please do not use in military or secret service organizations, or for illegal purposes (this is non-b
inding, these *** ignore laws and ethics anyway).

Hydra (https://github.com/vanhauser-thc/thc-hydra) starting at 2023-06-09 19:15:04
[WARNING] rdp servers often don't like many connections, use -t 1 or -t 4 to reduce the number of parallel connections and -W 1 or -W 3 to wait between connec
tion to allow the server to recover
[INFO] Reduced number of tasks to 4 (rdp does not like many parallel connections)
[WARNING] the rdp module is experimental. Please test, report - and if possible, fix.
[DATA] max 4 tasks per 1 server, overall 4 tasks, 55 login tries (l:11/p:5), ~14 tries per task
[DATA] attacking rdps://192.168.196.248:3389/
[VERBOSE] Resolving addresses ... [VERBOSE] resolving done
[ATTEMPT] target 192.168.196.248 - login "Luigi" - pass "Papal1963" - 1 of 55 [child 0] (0/0)
[ATTEMPT] target 192.168.196.248 - login "Luigi" - pass "Luigi=Papal1963" - 2 of 55 [child 1] (0/0)
[ATTEMPT] target 192.168.196.248 - login "Luigi" - pass "fireball" - 3 of 55 [child 2] (0/0)
[ATTEMPT] target 192.168.196.248 - login "Luigi" - pass "OathDeeplyReprieve91" - 4 of 55 [child 3] (0/0)
[ERROR] freerdp: The password has certainly expired and must be changed. (0x0002000f)
[ERROR] freerdp: The password has certainly expired and must be changed. (0x0002000f)
[VERBOSE] Disabled child 3 because of too many errors
[ERROR] freerdp: The password has certainly expired and must be changed. (0x0002000f)
[ERROR] freerdp: The password has certainly expired and must be changed. (0x0002000f)
[VERBOSE] Disabled child 0 because of too many errors
[VERBOSE] Disabled child 1 because of too many errors
[ERROR] all children were disabled due too many connection errors
0 of 1 target completed, 0 valid password found
Hydra (https://github.com/vanhauser-thc/thc-hydra) finished at 2023-06-09 19:15:06`

--- a/hydra-rdp.c
+++ b/hydra-rdp.c
@@ -76,6 +76,10 @@ int32_t start_rdp(char *ip, int32_t port, unsigned char options, char *miscptr, 
     // login failure
     hydra_completed_pair();
     break;
+  case 0x0002000f:
+    // login failure
+    hydra_completed_pair_skip();
+    break;
   case 0x0002000d:
     hydra_report(stderr,
                  "[%d][rdp] account on %s might be valid but account not "

